vm/actor/src/builtin/market/state.rs

```diff
@@ -0,0 +1,567 @@
+        states
+            .set(deal_id, state)
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+        self.states = states
+            .flush()
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+        Ok(TokenAmount::zero())
+    }
+    #[allow(dead_code)]
+    pub(super) fn delete_deal<BS>(
+        &mut self,
+        store: &BS,
+        deal_id: DealID,
+        deal: DealProposal,
+    ) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+    {
+        // let deal = self.must_get_deal(store, deal_id)?;
+
+        let mut proposals = Amt::<DealState, _>::load(&self.proposals, store)
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+        proposals
+            .delete(deal_id)
+            .map_err(|e| ActorError::new(ExitCode::ErrPlaceholder, e.into()))?;
+
+        let mut dbp = SetMultimap::from_root(store, &self.deal_ids_by_party)
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+        dbp.remove(&deal.client, deal_id)
+            .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e))?;
+
+        // Update roots of update
+        self.proposals = proposals
+            .flush()
+            .map_err(|e| ActorError::new(ExitCode::ErrPlaceholder, e.into()))?;
+        self.deal_ids_by_party = dbp
+            .root()
+            .map_err(|e| ActorError::new(ExitCode::ErrPlaceholder, e.into()))?;
+        Ok(())
+    }
+    #[allow(dead_code)]
+    pub(super) fn process_deal_init_timed_out<BS>(
+        &mut self,
+        store: &BS,
+        deal_id: DealID,
+        deal: DealProposal,
+        state: DealState,
+    ) -> Result<TokenAmount, ActorError>
+    where
+        BS: BlockStore,
+    {
+        assert!(state.sector_start_epoch.is_none());
+        self.unlock_balance(store, &deal.client, &deal.client_balance_requirement())?;
+
+        let amount_slashed =
+            collateral_penalty_for_deal_activation_missed(deal.provider_collateral.clone());
+        let amount_remainging = deal.provider_balance_requirement() - &amount_slashed;
+
+        self.slash_balance(store, &deal.provider, &amount_slashed)?;
+        self.unlock_balance(store, &deal.provider, &amount_remainging)?;
+        self.delete_deal(store, deal_id, deal)?;
+        Ok(amount_slashed)
+    }
+    #[allow(dead_code)]
+    pub(super) fn process_deal_expired<BS>(
+        &mut self,
+        store: &BS,
+        deal_id: DealID,
+        deal: DealProposal,
+        state: DealState,
+    ) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+    {
+        assert!(state.sector_start_epoch.is_some());
+
+        self.unlock_balance(store, &deal.provider, &deal.provider_collateral)?;
+        self.unlock_balance(store, &deal.client, &deal.client_collateral)?;
+
+        self.delete_deal(store, deal_id, deal)
+    }
+
```

Comment:
Although I don't see it used anywhere, [`generateStorageDealID`](https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/market/market_state.go#L220) is not included here.

Reply:
it's only used in the market actor, which is why I think I left it out initially. I added it now anyway though